c/progress/manager.cc

```diff
@@ -49,23 +49,25 @@ void progress_manager::finish_work(work* task, bool successfully) {
   xassert(pbar != nullptr);
   tasks.pop();
   if (successfully && tasks.empty()) {
+    std::lock_guard<std::mutex> lock(mutex);
     pbar->set_status_finished();
     delete pbar;
     pbar = nullptr;
   }
 }
 
 
-void progress_manager::update_view() const {
+void progress_manager::update_view() {
   xassert(dt::this_thread_index() == size_t(-1));
-  if (pbar) {
-    pbar->refresh();
-  }
+  if (!pbar) return;
+  std::lock_guard<std::mutex> lock(mutex);
+  pbar->refresh();
```

Comment:
`pbar` may still become null while the mutex waits to be locked

Reply:
Right, the same if true for `set_error_status()`, I will just swap mutex line with the nullptr check in these two methods.

Reply:
I'm not so sure, which is why I added a comment about the purpose of `mutex`. 
As far as I can see, `update_view()` is called from the monitor thread, and only from the monitor thread. All the other functions are called from the master thread only, presumably while the thread pool is sleeping (though the monitor thread is not in a thread pool, so its sleep cycle may be slightly out of sync).
The set_error_status() is interesting because it is called from `exception_to_python()`, but presumably we only enter python when the thread pool is already asleep.

These kind of assumptions tend to prove brittle over time though. Who knows, maybe in the next iteration we'll have the threadpool still working while the control is yielded to python -- in order to remember to verify these kind of assumptions, they should be encoded in form of asserts.